ReactNativeClient/lib/components/note-item.js

```diff
@@ -105,6 +105,17 @@ class NoteItemComponent extends Component {
 
 	render() {
 		const note = this.props.note ? this.props.note : {};
+
+		// this is to show only limited characters of title on note-list according to screen size
+		// the number 2 which is used to divide the height is here by testing screen sizes in different
+		// screen sizes emulator
+
+		const height = Dimensions.get('window').height;
+		const allowedLength = Math.round(height / 2);
+		if (note.title && note.title.length > allowedLength) {
+			note.title = `${note.title.substr(0,allowedLength)}...`;
+		}
```

Comment:
The note shouldn't be modified, that's a prop!! What if it's used somewhere else in the component or by the parent component? Your change to the title could even end up being saved to the database and that's the kind of bug that's very hard to figure out. If you need to change the title, create a copy of it in a const like `displayTitle` and use that.